cranelift/codegen/src/isa/aarch64/lower_inst.rs

```diff
@@ -1164,12 +1164,17 @@ pub(crate) fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
                 ctx.emit(alu_inst_imm12(alu_op, writable_zero_reg(), rn, rm));
                 ctx.emit(Inst::CondSet { cond, rd });
             } else {
-                if ty != I8X16 {
-                    return Err(CodegenError::Unsupported(format!(
-                        "unsupported simd type: {:?}",
-                        ty
-                    )));
-                }
+                match ty {
+                    I8X16 | I16X8 | I32X4 => {}
+                    _ => {
+                        // TODO: Should this return or assert? Shouldn't get here from wasm,
```

Comment:
I think the return is the best choice here -- in more recent PRs we've been trying to move toward `CodegenError`s rather than panics (partly this is out of a desire to allow clean failure of compilation in embeddings that might care about panics).